schemas/v3.0/schema.yaml

```diff
@@ -0,0 +1,1500 @@
+      readOnly:
+        type: boolean
+        default: false
+      writeOnly:
+        type: boolean
+        default: false
+      example: {}
+      externalDocs:
+        $ref: '#/definitions/ExternalDocumentation'
+      deprecated:
+        type: boolean
+        default: false
+      xml:
+        $ref: '#/definitions/XML'
+    patternProperties:
+      '^x-': {}
+    additionalProperties: false        
+  
+  Discriminator:
+    type: object
+    required:
+      - propertyName
+    properties:
+      propertyName:
+        type: string
+      mapping:
+        type: object
+        additionalProperties:
+          type: string
+
+  XML:
+    type: object
+    properties:
+      name:
+        type: string
+      namespace:
+        type: string
+        format: url
+      prefix:
+        type: string
+      attribute:
+        type: boolean
+        default: false
+      wrapped:
+        type: boolean
+        default: false
+    patternProperties:
+      '^x-': {}
+    additionalProperties: false
+
+  Response:
+    type: object
+    required:
+      - description
+    properties:
+      description:
+        type: string
+      headers:
+        additionalProperties:
+          oneOf:
+            - $ref: '#/definitions/Header'
+            - $ref: '#/definitions/Reference'
+      content:
+        type: object
+        additionalProperties:
+          $ref: '#/definitions/MediaType'
+      links:
+        type: object
+        additionalProperties:
+          oneOf:
+            - $ref: '#/definitions/Link'
+            - $ref: '#/definitions/Reference'
+    patternProperties:
+      '^x-': {}
+    additionalProperties: false
+
+  MediaType:
+    oneOf:
+      - $ref: '#/definitions/MediaTypeWithExample'
+      - $ref: '#/definitions/MediaTypeWithExamples'
+
+  MediaTypeWithExample:
+    type: object
+    properties:
+      schema:
+        oneOf:
+          - $ref: '#/definitions/Schema'
+          - $ref: '#/definitions/Reference'
+      example: {}
+      encoding:
+        type: object
+        additionalProperties:
+          $ref: '#/definitions/Encoding'
+    patternProperties:
+      '^x-': {}
+    additionalProperties: false 
+
+  MediaTypeWithExamples:
+    type: object
+    required:
+      - examples
+    properties:
+      schema:
+        oneOf:
+          - $ref: '#/definitions/Schema'
+          - $ref: '#/definitions/Reference'
+      examples:
+        type: object
+        additionalProperties:
+          oneOf:
+            - $ref: '#/definitions/Example'
+            - $ref: '#/definitions/Reference'
+      encoding:
+        type: object
+        additionalProperties:
+          $ref: '#/definitions/Encoding'
+    patternProperties:
+      '^x-': {}
+    additionalProperties: false 
+
+  Example:
+    type: object
+    properties:
+      summary:
+        type: string
+      description:
+        type: string
+      value: {}
+      externalValue:
+        type: string
+        format: uriref
```

Comment:
I think that the modern name of a `uriref` [is `uri-reference`](http://json-schema.org/latest/json-schema-validation.html#rfc.section.8.3.7). It might be better to switch over to that in the name of general correctness.

Reply:
OpenAPI 3.0.0 uses JSON schema Draft Wright 00 (aka Draft 5) so [uriref](https://tools.ietf.org/html/draft-wright-json-schema-validation-00#section-7.3.7) is correct, [uri-reference](http://json-schema.org/latest/json-schema-validation.html#rfc.section.8.3.7) is defined in Draft Wright 01 (aka JSON Schema Draft 6).

Reply:
Thanks Mike. I'd seen the [change in the changelog](http://json-schema.org/latest/json-schema-validation.html#rfc.appendix.B), but assumed maybe some tooling was outdated. Do you know what the rational for tying the spec and an older JSON schema version is? I guess because draft 6 introduces quite a few changes?

Reply:
@brandur ...because tying the spec to "whatever is the current version" would be impossible for tooling to deal with.  We had to lock on one version.

Reply:
@MikeRalphson it would be better to use `uri-reference` as formats are extensible.  It does not violate Draft 5, and is forward compatible with Draft 6.  There are actual Draft 6 validators in the wild (Draft 5 has never been directly implemented, and probably won't be).

Reply:
@handrews for better or worse, we're stuck with Draft 5 until at least OpenAPI 3.1.0 (or possibly 4.0 if it's deemed a breaking change). I think it would cause more confusion to use formats from Draft 6, and possibly result in those URI Reference properties not being validated at all. At least `ajv` can be convinced to operate in Draft 5 mode with a little setup. Of course there's no specific reason the schema for OpenAPI 3 has to itself be in Draft 5 just because OpenAPI 3 uses a modified version of Draft 5 internally, but again, it reduces confusion.

Reply:
@MikeRalphson I was under the impression that a lot of people use extended formats with OpenAPI?  Most validators that support formats, including Ajv allow registering additional ones.

If this is the only extended (from the draft-05 perspective) format being used, then yes, I agree it makes sense _not_ to use it.  But if it's not, I'd encourage folks to ignore `uriref` and use `uri-reference` instead, just as they would use `byte` or `password` or some other non-standard format.